test/components/attribute-picker/attribute-picker.test.js

```diff
@@ -0,0 +1,149 @@
+import '../../../src/components/attribute-picker/attribute-picker.js';
+import { expect, fixture, html, oneEvent, runConstructor, sendKeysElem } from '@brightspace-ui/testing';
+
+function createAttributeList(nameList) {
+	return nameList.map(name => ({
+		name,
+		value: `value_of_${name}`
+	}));
+}
+
+const assignableAttributeList = createAttributeList(['one', 'two', 'three', 'four', 'five', 'six']);
+const attributeList = createAttributeList(['one', 'two', 'three']);
+const shortAttributeList = createAttributeList(['one', 'two']);
+
+async function createComponent(attributeList, { allowFreeform = true, limit = 100 } = {}) {
+	const component = await fixture(
+		html`
+			<d2l-labs-attribute-picker
+				?allow-freeform=${allowFreeform}
+				.attributeList="${attributeList}"
+				.assignableAttributes="${assignableAttributeList}"
+				limit=${limit}>
+			</d2l-labs-attribute-picker>`
+	);
+
+	const listItems = component.shadowRoot.querySelectorAll('d2l-labs-attribute-picker-item');
+	for (const item of listItems) {
+		await item.updateComplete;
+	}
+
+	return component;
+}
+
+describe('attribute-picker', () => {
+	describe('constructor', () => {
+		it('should construct', () => {
+			runConstructor('d2l-labs-attribute-picker');
+		});
+	});
+
+	describe('interaction', () => {
+		it('should prevent unlisted if allow-freeform is disabled.', async() => {
+			const component = await createComponent(attributeList, { allowFreeform: false });
+			const input = component.shadowRoot.querySelector('input');
+			input.focus();
+
+			component._text = 'unlisted attribute';
```

Comment:
Might be worth trying to use `sendKeysElem` instead of focusing and setting the text programatically.